### PR TITLE
Improvement to check authenticator prerequisites before appending to redirection URL

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
When building redirection URL for next authentication step, check whether user satisfies all the prerequisites of the each authenticator and append only filtered authenticators to the URL.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java [hunk 1]
```diff
@@ -258,6 +259,8 @@ public static class Config {
         // Constant definitions for other QNames
         public static final String QNAME_AUTHENTICATION_ENDPOINT_URL = "AuthenticationEndpointURL";
         public static final String QNAME_AUTHENTICATION_ENDPOINT_RETRY_URL = "AuthenticationEndpointRetryURL";
+
```
Comment (Thisara-Welmilla) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java:262]: remove new line

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java [hunk 2]
```diff
@@ -445,6 +448,8 @@ public static class DefaultUrlContexts {
 
         public static final String AUTHENTICATION_ENDPOINT = "/authenticationendpoint/login.do";
         public static final String AUTHENTICATION_ENDPOINT_RETRY = "/authenticationendpoint/retry.do";
+
```
Comment (Thisara-Welmilla) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java:451]: remove new line

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java (hunk 1 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java]: Addressed

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java (hunk 2 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java]: Addressed

**Review by DMHP — APPROVED** [review]

**Pull request merged**